### PR TITLE
Firmware: Improvements in getting firmware directly from macOS on Linux

### PR DESCRIPTION
**Pull request by AdityaGarg8**

### CODE REVIEW

**Comment by AdityaGarg8** [comment]
@sharpenedblade @soopyc please test.

File: docs/tools/firmware.sh [hunk 1]
```diff
@@ -577,49 +577,58 @@ case "$os" in
...
-				dmg2img_check
 				# Load the apfs driver, and install if missing
-				sudo modprobe apfs >/dev/null 2>&1 || apfs_install
+				sudo modprobe ${verbose} apfs 2>/dev/null || apfs_install
 				unmount_macos_and_cleanup () {
 					sudo rm -r ${verbose} ${workdir} || true
-					sudo umount ${verbose} ${loopdevice} || true
-					sudo umount ${verbose} ${macosvol} || true
-					sudo rm -r ${verbose} ${imgdir} || true
+					for i in 0 1 2 3 4 5
```
Comment (Redecorating) [inline docs/tools/firmware.sh:585]: If the user has made extra apfs volumes could this cause issues? I think i have a couple extra volumes so I'll test it when I boot to Linux in a minute.

Comment (AdityaGarg8) [reply docs/tools/firmware.sh:585]: I had tried making extra Volumes, they get the number from 6. Still, it would be better to test it from your side.

Comment (Redecorating) [reply docs/tools/firmware.sh:585]: hmm seems to work fine for me

**Review by Redecorating — APPROVED** [review]

File: docs/tools/firmware.sh (hunk 1 above)
Comment (AdityaGarg8) [reply docs/tools/firmware.sh:585]: > hmm seems to work fine for me

Any error you encountered if you run it with -v

Comment (Redecorating) [reply docs/tools/firmware.sh:585]: just errors about some of the apfs volumes not being mountable but that's expected

Comment (AdityaGarg8) [reply docs/tools/firmware.sh:585]: Thanks.

**Pull request merged**